docs/reference/contracts/apis/managed.md

```diff
@@ -0,0 +1,294 @@
+# Managed Pools
+
+## BPT Supply
+
+### `getActualSupply`
+
+```solidity
+getActualSupply() 
+returns (uint256)
+```
+
+Returns the effective BPT Supply. Implemented in `ManagedPoolSettings`. 
+
+::: warning
+Since this function reads balances directly from the Vault, it is potentially subject to manipulation via reentrancy. To call this function safely, attempt to trigger the reentrancy guard in the Vault by calling a non-reentrant function before calling `getActualSupply`.
+:::
+
+## Swap Fees
+
+### `getSwapFeePercentage`
+
+```
+getSwapFeePercentage() 
+returns (uint256)
+```
+
+Returns the current value of the swap fee percentage. Implemented in `ManagedPoolSettings`.
+
+### `getGradualSwapFeeUpdateParams`
+```solidity
+getGradualSwapFeeUpdateParams()
+returns (uint256 startTime,
+         uint256 endTime,
+         uint256 startSwapFeePercentage,
+         uint256 endSwapFeePercentage)
+```
+
+Returns the current gradual swap fee update parameters. Implemented in `ManagedPoolSettings`.
+
+### `updateSwapFeeGradually`
+
+```solidity
+updateSwapFeeGradually(uint256 startTime,
+                       uint256 endTime,
+                       uint256 startSwapFeePercentage,
+                       uint256 endSwapFeePercentage)
+
+emits GradualSwapFeeUpdateScheduled(uint256 startTime,
+                                    uint256 endTime,
+                                    uint256 startSwapFeePercentage,
+                                    uint256 endSwapFeePercentage)
+```
+
+Schedules a gradual update in swap fees at the block timestamp provided in `startTime`, up until `endTime`. This is a permissioned function that can only be called by an authorized address. If the pool is paused, the call will revert. Implemented in `ManagedPoolSettings`.
+
+## Weights
+
+### `getNormalizedWeights`
+
+```solidity
+getNormalizedWeights() 
+returns (uint256[] memory)
+```
+
+Returns the normalized weights in the same order as the pool's tokens. Implemented in `ManagedPoolSettings`.
```

Comment:
This is generally true. To be more specific, it might be helpful to add that if the weights are undergoing a gradual update, it is the weights at the time of the last confirmed block